ntoskrnl/cc/copy.c

```diff
@@ -652,15 +652,13 @@ CcCopyWrite (
             Buffer = (PVOID)((ULONG_PTR)Buffer + VacbLength);
             CurrentOffset += VacbLength;
 
-            /* Tell Mm */
-            Status = MmMakePagesDirty(NULL, Add2Ptr(Vacb->BaseAddress, VacbOffset), VacbLength);
+            Status = CcpMarkCacheDataDirty(Vacb, VacbOffset, VacbLength);
             if (!NT_SUCCESS(Status))
                 ExRaiseStatus(Status);
         }
         _SEH2_FINALLY
         {
-            /* Do not mark the VACB as dirty if an exception was raised */
-            CcRosReleaseVacb(SharedCacheMap, Vacb, !_SEH2_AbnormalTermination(), FALSE);
+            CcRosReleaseVacb(SharedCacheMap, Vacb, FALSE);
         }
         _SEH2_END;
```

Comment:
What if part of the write succeeded and then an exception happens? Shouldn't the VACB always be marked dirty?

Reply:
There is no reason for CC to mark the VACB dirty or update the dirty range of the VACB if `MmMakePagesDirty` failed because MM won't flush the pages anyway.